src/app/pages/ProposalDetailsPage/hooks.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  List,
+  useGetConsensusProposalsProposalIdVotes,
+  useGetConsensusValidators,
+} from '../../../oasis-nexus/api'
+import { Network } from '../../../types/network'
+import {
+  ExtendedVote,
+  getFilterForVoteType,
+  getRandomVote,
+  ProposalVoteValue,
+  VoteFilter,
+  VoteType,
+} from '../../../types/vote'
+import { useClientSidePagination } from '../../components/Table/useClientSidePagination'
+import { NUMBER_OF_ITEMS_ON_SEPARATE_PAGE } from '../../config'
+import { useStringInUrl } from '../../hooks/useStringInUrl'
+
+export type AllVotesData = List & {
+  isLoading: boolean
+  isError: boolean
+  loadedVotes: ExtendedVote[]
+}
+
+const DEBUG_MODE = true // TODO disable debug mode before merging
+
+const voteTable: Record<string, ProposalVoteValue> = {}
+
+const getRandomVoteFor = (address: string) => {
+  const storedVote = voteTable[address]
+  if (storedVote) return storedVote
+  const newVote = getRandomVote()
+  voteTable[address] = newVote
+  return newVote
+}
+
+const useValidatorMap = (network: Network) => {
+  const { data, isLoading, isError } = useGetConsensusValidators(network)
+  return {
+    isLoading,
+    isError,
+    map: (data as any)?.data.map ?? new Map(),
+  }
+}
+
+export const useAllVotes = (network: Network, proposalId: number): AllVotesData => {
+  const query = useGetConsensusProposalsProposalIdVotes(network, proposalId)
+  const {
+    map: validators,
+    isLoading: areValidatorsLoading,
+    isError: haveValidatorsFailed,
+  } = useValidatorMap(network)
+  const { isLoading, isError, data } = query
+
+  const extendedVotes = (data?.data.votes || []).map(
+    (vote, index): ExtendedVote => ({
+      ...vote,
+      index: index + 1,
```

Comment:
(keep index zero-based and +1 in display)

Reply:
Updated accordingly